sdk/appconfiguration/azure-data-appconfiguration/inc/azure/data/appconfiguration/configuration_client_models.hpp

```diff
@@ -0,0 +1,614 @@
+    Nullable<std::vector<Error>> Details;
+    InnerError Innererror;
+  };
+
+  struct OperationDetails final
+  {
+    std::string Id;
+    OperationState Status;
+    AppConfiguration::Error Error;
+  };
+
+  struct CreateSnapshotResult final
+  {
+    std::string Name;
+    Nullable<SnapshotStatus> Status;
+    std::vector<KeyValueFilter> Filters;
+    Nullable<AppConfiguration::CompositionType> CompositionType;
+    Nullable<std::string> Created;
+    Nullable<std::string> Expires;
+    Nullable<std::int64_t> RetentionPeriod;
+    Nullable<std::int64_t> Size;
+    Nullable<std::int64_t> ItemsCount;
+    Nullable<std::map<std::string, std::string>> Tags;
+    Nullable<std::string> Etag;
+    std::string SyncToken;
+    std::string ETag;
+    std::string Link;
+    Core::Url OperationLocation;
+    CreateSnapshotResponseContentType ContentType;
+  };
+
+  struct UpdateSnapshotResult final
+  {
+    std::string Name;
+    Nullable<SnapshotStatus> Status;
+    std::vector<KeyValueFilter> Filters;
+    Nullable<AppConfiguration::CompositionType> CompositionType;
+    Nullable<std::string> Created;
+    Nullable<std::string> Expires;
+    Nullable<std::int64_t> RetentionPeriod;
+    Nullable<std::int64_t> Size;
+    Nullable<std::int64_t> ItemsCount;
+    Nullable<std::map<std::string, std::string>> Tags;
+    Nullable<std::string> Etag;
+    std::string SyncToken;
+    std::string ETag;
+    std::string Link;
+    UpdateSnapshotResponseContentType ContentType;
+  };
+
+  struct SnapshotUpdateParameters final
+  {
+    Nullable<SnapshotStatus> Status;
+  };
+
+  struct CheckSnapshotResult final
+  {
+    std::string SyncToken;
+    std::string ETag;
+    std::string Link;
+  };
+
+  struct Label final
+  {
+    Nullable<std::string> Name;
+  };
+
+  struct CheckLabelsResult final
+  {
+    std::string SyncToken;
+  };
+
+  struct PutLockResult final
+  {
+    std::string Key;
+    Nullable<std::string> Label;
+    Nullable<std::string> ContentType;
+    Nullable<std::string> Value;
+    Nullable<std::string> LastModified;
+    Nullable<std::map<std::string, std::string>> Tags;
+    Nullable<bool> Locked;
+    Nullable<std::string> Etag;
+    std::string SyncToken;
+    std::string ETag;
+  };
+
+  struct DeleteLockResult final
+  {
+    std::string Key;
+    Nullable<std::string> Label;
+    Nullable<std::string> ContentType;
+    Nullable<std::string> Value;
+    Nullable<std::string> LastModified;
+    Nullable<std::map<std::string, std::string>> Tags;
+    Nullable<bool> Locked;
+    Nullable<std::string> Etag;
+    std::string SyncToken;
+    std::string ETag;
+  };
+
+  struct CheckRevisionsResult final
+  {
+    std::string SyncToken;
+    std::string ETag;
+  };
+
+  class GetKeysPagedResponse final : public Core::PagedResponse<GetKeysPagedResponse> {
+    friend class ConfigurationClient;
+    friend class Core::PagedResponse<GetKeysPagedResponse>;
+
+  private:
+    std::shared_ptr<ConfigurationClient> m_client;
```

Comment:
Maybe. I am not sure.

We'll need to evaluate the generated pattern for PagedResponse compare to what we expect to generate. For example, we shouldn't have the operation specific options as a member field, such as `GetKeysOptions m_options;`

Our existing PagedResponse pattern **does** hold a shared_ptr to the client, though it is created using `make_unique` rather than `make_shared`.
https://github.com/Azure/autorest.cpp/blob/9865d599f9e539548f9b99f50649400e88630892/packages/typespec-cpp/samples/appconfig/generated/src/rest_client.cpp#L225-L226

https://github.com/Azure/azure-sdk-for-cpp/blob/3687136d04cc8e425cb368ce879a1686834b17c9/sdk/keyvault/azure-security-keyvault-certificates/src/certificate_client.cpp#L400-L401
https://github.com/Azure/azure-sdk-for-cpp/blob/3687136d04cc8e425cb368ce879a1686834b17c9/sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp#L1307

Reply:
https://github.com/Azure/autorest.cpp/issues/427